File: 1524-string-matching-in-an-array/1524-string-matching-in-an-array.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        
        res = []

        for i, word in enumerate(words):
            for j, other in enumerate(words):
                if i != j and word in other:
                    res.append(word)
                    break  

        return res


                 

File: 1524-string-matching-in-an-array/1524-string-matching-in-an-array.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var stringMatching = function(words) {
    
    let result = []

    words.forEach((word, i) => {
    words.forEach((other, j) => {
        if (i !== j && other.includes(word)) {
            if (!result.includes(word)){
                result.push(word)}
            return; // stop checking further
        }
    });
});

    return result
};